### PR TITLE
add active lock reason to PR

### LINKED ISSUE #2248: Issue active_lock_reason missing
Hello,

I am trying to get the value of

- `active_lock_reason` JSON 
- `activeLockReason` GraphQL

on the issue object.

Ive searched the C# source code but I do not think this property is accessible?

### PR DESCRIPTION
**Pull request by notauserx**
add active lock reason to PR. 
Since every pull request is an issue and the lock and unlock are on the IIssuesClient, updates the docs to reflect the design.
Fixes #2248

### CODE REVIEW

**Comment by notauserx** [comment]
I have extracted the lock/unlock methods from IIssuesClient to a new ILockUnlockClient, which is available in both the IIssuesClient and IPullRequestClient, so both clients can use the ILockUnlockClient.

I need a review for this PR and any feedback. I have intentionally left out the refactoring on the Reactive project, I will get it that once I get a review.

**Comment by JonruAlveus** [comment]
Hi @notauserx, thanks for this contribution ♥️! What you’ve done looks great, there’s just a couple of points:

- Please check the pr for warnings raised during the build as we’re trying to keep the warnings to none if possible
- It looks like the build is failing, but I think you’ve got that covered by the upcoming Reactive changes
- If you have time (and the desire!), please could you add/update the readme md files in the root/docs folder and/or any related linqpad samples?

**Comment by notauserx** [comment]
Thanks @JonruAlveus for the feedback. I will also update the relevant readme and the linqpad samples in the PR.

**Comment by notauserx** [comment]
@JonruAlveus I have updated the docs in the Issues.md. I did not find any relevant linqpad samples to update. Please let me know what you think.

File: Octokit.Tests.Integration/Clients/PullRequestsClientTests.cs [hunk 1]
```diff
@@ -653,6 +653,25 @@ public async Task CanClose()
...
+
+        await _fixture.LockUnlock.Lock(Helper.UserName, _context.RepositoryName, pullRequest.Number, LockReason.OffTopic);
+        var retrived = await _fixture.Get(Helper.UserName, _context.RepositoryName, pullRequest.Number);
+
+        Assert.Equal(retrived.ActiveLockReason, LockReason.OffTopic);
+
+        await _fixture.LockUnlock.Unlock(Helper.UserName, _context.RepositoryName, pullRequest.Number);
+        var unlocked = await _fixture.Get(Helper.UserName, _context.RepositoryName, pullRequest.Number);
+
+        Assert.Equal(unlocked.ActiveLockReason, null);
```
Comment (JonruAlveus) [inline Octokit.Tests.Integration/Clients/PullRequestsClientTests.cs:672]: Please could you address the warnings just below here?

**Review by JonruAlveus — CHANGES_REQUESTED** [review]
Super happy with everything you’ve done here, if you can fix up the warnings I’d be happy to approve for merge!

File: Octokit.Tests.Integration/Clients/PullRequestsClientTests.cs (hunk 1 above)
Comment (notauserx) [reply Octokit.Tests.Integration/Clients/PullRequestsClientTests.cs]: Thanks for catching that. Updated the PR with Assert.Null.

**Review by JonruAlveus — APPROVED** [review]

**Comment by JonruAlveus** [comment]
release_notes: Reworks how Locking and Unlocking works on Issues and adds Locking and Unlocking to Pull Requests

**Pull request merged**